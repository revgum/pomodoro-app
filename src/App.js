// @flow
import React from "react";
import "./App.css";
import Timer from "./Timer";
import PomodoroList from "./PomodoroList";
import type { AppState, TimerState } from "./types";
import { destroy, fetchAll, save } from "./api";

// material-ui
import { withStyles } from "material-ui/styles";
import Paper from "material-ui/Paper";
import Grid from "material-ui/Grid";

type Props = {
  classes: any
};

const styles = theme => ({
  root: {
    flexGrow: 1
  },
  paper: {
    padding: theme.spacing.unit * 2,
    textAlign: "center",
    color: theme.palette.text.secondary
  }
});

class App extends React.Component<Props, AppState> {
  static defaultProps = {};

  state: AppState = {
    pomodoros: [],
    timer: {
      minutes: 25,
      seconds: 0
    }
  };

  componentDidMount() {
    fetchAll()
      .then(pomodoros => this.setState({ pomodoros }))
      .catch(error => console.error("ERROR", error));
  }

  render() {
    return (
      <div className={this.props.classes.root}>
        <Grid container spacing={24}>
          <Grid item sm={12} md={8}>
            <Paper className={this.props.classes.paper}>
              <Timer
                {...this.state.timer}
                completeSession={this.completeSession.bind(this)}
              />
            </Paper>
          </Grid>
          <Grid item sm={12} md={4}>
            <PomodoroList
              deletePomodoro={this.deletePomodoro.bind(this)}
              pomodoros={this.state.pomodoros}
            />
          </Grid>
        </Grid>
      </div>
    );
  }

  completeSession = (timer_state: TimerState) => {
    // TODO: Add some Notification component and related state for success/error to the UI
    save(timer_state)
      .then(pomodoro => {
        this.setState({ pomodoros: [pomodoro, ...this.state.pomodoros] });
      })
      .catch(error => console.error("ERROR", error));
  };

  deletePomodoro = (timer_state: TimerState) => {
    destroy(timer_state)
      .then(pomodoro => {
        fetchAll()
          .then(pomodoros => this.setState({ pomodoros }))
          .catch(error => console.error("ERROR", error));
      })
      .catch(error => console.error("ERROR", error));
  };
}

export default withStyles(styles)(App);
