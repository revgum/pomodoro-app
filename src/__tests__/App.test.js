import React from "react";
import ReactDOM from "react-dom";
import App from "../App";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

// Basic test full rendering
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// Shallow rendering test
it("renders without crashing", () => {
  shallow(<App />);
});

// Jest matcher test
it("renders without crashing", () => {
  const wrapper = shallow(<App />);
  const welcome = <h1 className="App-Title">Welcome to React</h1>;
  expect(wrapper.contains(welcome)).toEqual(true);
  // or use jest-enzyme matchers
  expect(wrapper).toContainReact(welcome);
});

it("renders the snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
