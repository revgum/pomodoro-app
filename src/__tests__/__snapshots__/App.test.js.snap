// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> captures a snapshot 1`] = `
<div
  className="App-root-1"
>
  <div
    className="MuiGrid-container-3 MuiGrid-spacing-xs-24-26"
  >
    <div
      className="MuiGrid-item-4 MuiGrid-grid-sm-12-53 MuiGrid-grid-md-8-62"
    >
      <div
        className="MuiPaper-root-93 MuiPaper-elevation2-97 MuiPaper-rounded-94 App-paper-2"
      >
        <div
          className="timer"
        >
          <div
            className="remaining"
          >
            25:00
          </div>
          <div
            aria-valuenow={100}
            className="MuiLinearProgress-root-120 MuiLinearProgress-colorPrimary-121"
            role="progressbar"
          >
            <div
              className="MuiLinearProgress-bar-128 MuiLinearProgress-barColorPrimary-129 MuiLinearProgress-bar1Determinate-133"
              style={
                Object {
                  "transform": "scaleX(1)",
                }
              }
            />
          </div>
          <div
            className="timer-controls"
          >
            <button
              aria-label="Stop"
              className="MuiButtonBase-root-144 MuiIconButton-root-138 Controls-button-136 stop"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-143"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-147"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M12 6c3.87 0 7 3.13 7 7 0 .84-.16 1.65-.43 2.4l1.52 1.52c.58-1.19.91-2.51.91-3.92 0-4.97-4.03-9-9-9-1.41 0-2.73.33-3.92.91L9.6 6.43C10.35 6.16 11.16 6 12 6zm10-.28l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM2.92 2.29L1.65 3.57 2.98 4.9l-1.11.93 1.42 1.42 1.11-.94.8.8C3.83 8.69 3 10.75 3 13c0 4.97 4.02 9 9 9 2.25 0 4.31-.83 5.89-2.2l2.2 2.2 1.27-1.27L3.89 3.27l-.97-.98zm13.55 16.1C15.26 19.39 13.7 20 12 20c-3.87 0-7-3.13-7-7 0-1.7.61-3.26 1.61-4.47l9.86 9.86zM8.02 3.28L6.6 1.86l-.86.71 1.42 1.42.86-.71z"
                    />
                  </g>
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-153"
              />
            </button>
            <button
              aria-label="Pause"
              className="MuiButtonBase-root-144 MuiIconButton-root-138 Controls-button-136 pause"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-143"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-147"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M9 16h2V8H9v8zm3-14C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm1-4h2V8h-2v8z"
                    />
                  </g>
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-153"
              />
            </button>
            <button
              aria-label="Restart"
              className="MuiButtonBase-root-144 MuiIconButton-root-138 Controls-button-136 reset"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-143"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-147"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M7 7h10v3l4-4-4-4v3H5v6h2V7zm10 10H7v-3l-4 4 4 4v-3h12v-6h-2v4zm-4-2V9h-1l-2 1v1h1.5v4H13z"
                    />
                  </g>
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-153"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-4 MuiGrid-grid-sm-12-53 MuiGrid-grid-md-4-58"
    >
      <div
        className="PomodoroList-listContainer-162 pomodoro-list"
      >
        <ul
          className="PomodoroList-list-163"
        >
          <li
            onClick={[Function]}
          >
            <div
              className="MuiGrid-container-3 MuiGrid-align-items-xs-flex-end-13 PomodoroList-root-160"
            >
              <div
                className="MuiGrid-item-4"
              >
                <h2
                  className="PomodoroList-listHeader-164"
                >
                  Pomodoros
                </h2>
              </div>
              <div
                className="MuiGrid-item-4"
              />
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
